src/controllers/communityController.js

```diff
@@ -93,35 +94,43 @@ const deleteCommunity = async (req, res) => {
     );
 };
 
-const getAllTripsOfCommunity = async (req, res) => {
-
-    client.query(`
-SELECT trips.*
-FROM trips
-INNER JOIN community_trips ON trips.trip_id = community_trips.trip_id
-WHERE community_trips.community_id = $1;
-`
-        , [req.params.community_id],
-
-        function(error, results) {
-            if (!error && results.rows.length != 0) {
-                res.status(201).send(results);
-            } else if (results.rows.length == 0) {
+const getAllTripsOfCommunity = async (req, res, next) => {
+    // Collect unique dates, origins, and destinations
+    const tripIds = new Set();
+    const community_id = req.params.community_id;
+    
+    try {
+
+        let sqlQuery = `
+        SELECT DISTINCT CT.trip_id
```

Comment:
Here, we are getting the trip ids in one query, then passing it to the queryTrips. Which once again performs a query to get all trips. After that you filter those trips against these trip ids. 
instead we can simply construct this one query to achieve the parameters. 
The existing query directly fetches all the trip rows in the community, we can change the query params with minimal effort.
Also, I do not understand why you need to fetch all users to ultimately only consider the trips relevant to the user, while you directly have the trips revelant to the user from this query.

Reply:
Ah! I saw that now I did a mistake actually I have to use the sent tripIds not to again perform a query to get all the trips in the queryTrips function I will fix it.

Reply:
Actually what is happening we are getting all the tripIds relevant to a community from the getAllTripsOfCommunity function then we are sending these tripIds to the queryTrips function where now this function is getting all the information of the tripIds and then different filters are applied.

Reply:
lets leave the filters for later, for now do not change this function.

Reply:
Revert the changes in that function and make a commit, I will merge it